libs/server/Storage/Functions/MainStore/RMWMethods.cs

```diff
@@ -911,14 +911,16 @@ public bool NeedCopyUpdate(ref SpanByte key, ref RawStringInput input, ref SpanB
                 case RespCommand.DELIFGREATER:
                     if (rmwInfo.RecordInfo.ETag)
                         EtagState.SetValsForRecordWithEtag(ref functionsState.etagState, ref oldValue);
+
                     long etagFromClient = input.parseState.GetLong(0);
-                    if (etagFromClient <= functionsState.etagState.etag)
+                    if (etagFromClient > functionsState.etagState.etag)
                     {
-                        EtagState.ResetState(ref functionsState.etagState);
-                        return false;
+                        rmwInfo.Action = RMWAction.ExpireAndStop;
```

Comment:
Add a comment here explaining why there's no "true" return: Either we ExpireAndStop (in which case we must return false) or there is no CopyUpdate to be done (so again we must return false).